node/external/transactionAPI/apiTransactionResults.go

```diff
@@ -0,0 +1,222 @@
+	scrHashesEpoch []*dblookupext.ScResultsHashesAndEpoch,
+) {
+	for _, scrHashesE := range scrHashesEpoch {
+		for _, scrHash := range scrHashesE.ScResultsHashes {
+			scr, err := arp.getScrFromStorage(scrHash, scrHashesE.Epoch)
+			if err != nil {
+				log.Warn("putSmartContractResultsInTransaction cannot get result from storage",
+					"hash", hex.EncodeToString(scrHash),
+					"error", err.Error())
+				continue
+			}
+
+			scrAPI := arp.adaptSmartContractResult(scrHash, scr)
+			arp.putLogsInSCR(scrHash, scrHashesE.Epoch, scrAPI)
+
+			tx.SmartContractResults = append(tx.SmartContractResults, scrAPI)
+		}
+	}
+
+	statusFilters := filters.NewStatusFilters(arp.selfShardID)
+	statusFilters.SetStatusIfIsFailedESDTTransfer(tx)
+}
+
+func (arp *apiTransactionResultsProcessor) putLogsInTransaction(hash []byte, tx *transaction.ApiTransactionResult, epoch uint32) {
+	logsAndEvents, err := arp.getLogsAndEvents(hash, epoch)
+	if err != nil || logsAndEvents == nil {
+		return
+	}
+
+	logsAPI := arp.prepareLogsAndEvents(logsAndEvents)
+	tx.Logs = logsAPI
+}
+
+func (arp *apiTransactionResultsProcessor) putLogsInSCR(scrHash []byte, epoch uint32, scr *transaction.ApiSmartContractResult) {
+	logsAndEvents, err := arp.getLogsAndEvents(scrHash, epoch)
+	if err != nil {
+		return
+	}
+
+	logsAPI := arp.prepareLogsAndEvents(logsAndEvents)
+	scr.Logs = logsAPI
+}
+
+func (arp *apiTransactionResultsProcessor) getScrFromStorage(hash []byte, epoch uint32) (*smartContractResult.SmartContractResult, error) {
+	unsignedTxsStorer := arp.storageService.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	scrBytes, err := unsignedTxsStorer.GetFromEpoch(hash, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	scr := &smartContractResult.SmartContractResult{}
+	err = arp.marshalizer.Unmarshal(scr, scrBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return scr, nil
+}
+
+func (arp *apiTransactionResultsProcessor) adaptSmartContractResult(scrHash []byte, scr *smartContractResult.SmartContractResult) *transaction.ApiSmartContractResult {
+	apiSCR := &transaction.ApiSmartContractResult{
+		Hash:           hex.EncodeToString(scrHash),
+		Nonce:          scr.Nonce,
+		Value:          scr.Value,
+		RelayedValue:   scr.RelayedValue,
+		Code:           string(scr.Code),
+		Data:           string(scr.Data),
+		PrevTxHash:     hex.EncodeToString(scr.PrevTxHash),
+		OriginalTxHash: hex.EncodeToString(scr.OriginalTxHash),
+		GasLimit:       scr.GasLimit,
+		GasPrice:       scr.GasPrice,
+		CallType:       scr.CallType,
+		CodeMetadata:   string(scr.CodeMetadata),
+		ReturnMessage:  string(scr.ReturnMessage),
+	}
+
+	if len(scr.SndAddr) != 0 {
+		apiSCR.SndAddr = arp.addressPubKeyConverter.Encode(scr.SndAddr)
+	}
+
+	if len(scr.RcvAddr) != 0 {
+		apiSCR.RcvAddr = arp.addressPubKeyConverter.Encode(scr.RcvAddr)
+	}
+
+	if len(scr.RelayerAddr) != 0 {
+		apiSCR.RelayerAddr = arp.addressPubKeyConverter.Encode(scr.RelayerAddr)
+	}
+
+	if len(scr.OriginalSender) != 0 {
```

Comment:
can be compared with `addressPubKeyConverter.Len()` instead of `0`

Reply:
done.